pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,112 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing_extensions import Literal, get_args
+
+import numpy as np
+import qutip
+
+
+NOISE_TYPES = Literal['doppler', 'amplitude', 'SPAM', 'dephasing']
+mass = 1.45e-25
+kb = 1.38e-23
+keff = 8.7
+
+
+@dataclass
+class SimConfig:
+    """Include additional parameters to simulation.
+
+        The user chooses the settings when creating a SimConfig. Settings
+        shouldn't be changed after instantiation of a SimConfig.
+
+        Keyword arguments:
+            noise (list[NOISE_TYPES]): Types of noises to be used in the
+                simulation. Choose among:
+                -   'dephasing': Random phase (Z) flip
+                -   'doppler': Noisy doppler runs
+                -   'amplitude': Noisy gaussian beam
+                -   'SPAM': SPAM errors. Adds:
+                    --  eta: Probability of each atom to be badly prepared
+                    --  epsilon: Probability of false positives
+                    --  epsilon_prime: Probability of false negatives.
+            runs (int): Number of runs needed : each run draws a new random
+                noise.
+            samples_per_run (int): Number of samples per noisy run.
+                Useful for cutting down on computing time, but unrealistic.
+            temperature (float): Temperature, set in muK, of the Rydberg array.
+                Also sets the standard deviation of the speed of the atoms.
+            laser_waist (float): Waist of the gaussian laser in global pulses.
+            solver_options (qutip.Options): options for the qutip solver.
+        """
+    noise: list[NOISE_TYPES] = field(default_factory=list)
+    runs: int = 15
+    samples_per_run: int = 5
+    temperature: float = 50.
+    laser_waist: float = 175.
+    eta: float = 0.005
+    epsilon: float = 0.01
+    epsilon_prime: float = 0.05
+    solver_options: qutip.Options = qutip.Options(max_step=5)
+
+    def __post_init__(self):
+        self._process_temperature()
+        self._check_noise_types()
+        self.spam_dict = {'eta': self.eta, 'epsilon': self.epsilon,
+                          'epsilon_prime': self.epsilon_prime}
+        self._check_spam_dict()
+        self._calc_sigma_doppler()
+
+    def __str__(self) -> str:
+        s = f"""Options:
+----------
+Noise types:         {self.noise}
+Spam dictionary:     {self.spam_dict}
+Temperature:         {self.temperature}K
+Number of runs:      {self.runs}
+Samples per runs:    {self.samples_per_run}
+Laser waist:         {self.laser_waist}μm
+
+Solver Options:
+{str(self.solver_options)[10:-1]}"""
```

Comment:
nit: The indentation looks weird like this... I would rather have each line on a list and then joined. See an example  in `Device._specs()`.